api/app/Support/PeriodSupport.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace App\Support;
+
+use App\Exceptions\UndefinedPeriodException;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Str;
+use stdClass;
+
+class PeriodSupport
+{
+    /**
+     * Dynamically retrieve period
+     *
+     * @param string $period
+     *
+     * @return stdClass
+     *
+     * @throws UndefinedPeriodException
+     *
+     * @author Roy Freij <roy@bsbip.com>
+     * @version 1.0.0
+     */
+    public static function retrieve(string $period): stdClass
+    {
+        $period = Str::camel($period);
+
+        if (method_exists(self::class, $period)) {
+            return self::$period();
+        }
+
+        throw (new UndefinedPeriodException($period));
+    }
+
+    /**
+     * Retrieve period for current month
+     *
+     * @return stdClass
+     *
+     * @author Roy Freij <roy@bsbip.com>
+     * @version 1.0.0
+     */
+    public static function currentMonth(): stdClass
+    {
+        return (object) [
+            'from' => Carbon::now()->startOfMonth(),
```

Comment:
A function startOf and a function endOf exists, with the possiblity to pass as string as argument. Therefore, it should be possible to make this dynamic for each period (month, week, etc.).

Reply:
Those functions are added in Carbon 2.0, since laravel 5.5 doesn't support that we can't use it. 
Laravel supports Carbon 2.0 starting from 5.8

![image](https://user-images.githubusercontent.com/15109384/63632882-5d2d4d00-c63f-11e9-99a6-10fd353df992.png)

Reply:
It seems to be possible to use Carbon 2 with Laravel 5.5 by using https://github.com/kylekatarnls/laravel-carbon-2, but let us refactor it after the API is upgraded to a newer Laravel version.